server/lib/currency.ts

```diff
@@ -150,13 +163,13 @@ export async function fetchFxRates(
     }
   }
 
-  // In case of error or if Fixer API is not configured, fallback to DB/mock values
-  if (!isLiveEnv) {
+  if (useMockRate) {
     const ratesValues = toCurrencies.map(() => 1.1);
     return zipObject(toCurrencies, ratesValues);
-  } else {
-    return getRatesFromDb(fromCurrency, toCurrencies, date);
   }
+
+  // As a fallback, try to fetch the rates from the DB
+  return getRatesFromDb(fromCurrency, toCurrencies, date);
```

Comment:
🍨 A possible optimization would be checking the DB first for a fresh enough date and storing the recently fetched rate from Fixer every time we don't find it.